src/test/java/core/basesyntax/operation/SupplyOperationHandlerTest.java

```diff
@@ -0,0 +1,47 @@
+package core.basesyntax.operation;
+
+import static org.junit.Assert.assertEquals;
+
+import core.basesyntax.FruitTransaction;
+import core.basesyntax.db.StorageService;
+import core.basesyntax.db.StorageServiceImpl;
+import java.util.HashMap;
+import java.util.Map;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+class SupplyOperationHandlerTest {
+    private StorageService storageService;
+    private SupplyOperationHandler supplyOperationHandler;
+
+    @BeforeEach
+    public void setUp() {
+        storageService = new StorageServiceImpl();
+        supplyOperationHandler = new SupplyOperationHandler(storageService);
+    }
+
+    @Test
+    public void handle_Ok() {
+        FruitTransaction transaction = new FruitTransaction();
+        transaction.setFruit("apple");
+        transaction.setAmount(5);
+
+        Map<String, Integer> storage = new HashMap<>();
+
+        supplyOperationHandler.handle(transaction, storage);
+        assertEquals(5, storageService.getStorage().get("apple").intValue());
+    }
+
+    @Test
+    public void handle_negativeAmount_NotOk() {
+        FruitTransaction transaction = new FruitTransaction();
+        transaction.setFruit("apple");
+        transaction.setAmount(-3);
+
+        Map<String, Integer> storage = new HashMap<>();
+
+        supplyOperationHandler.handle(transaction, storage);
+        assertEquals(-3, storageService.getStorage()
+                .getOrDefault("apple", 0).intValue());
+    }
```

Comment:
wait, shouldn't we throw exception in such case? why do we allow negative amounts